controllers/topology/internal/mergepatch/mergepatch.go

```diff
@@ -58,22 +58,33 @@ func NewHelper(original, modified client.Object, c client.Client, opts ...Helper
 		{"spec"},
```

Comment:
Should we add a note that the managedField logic is only correct if the whole "spec" tree is part of `allowedPaths`?

I know we probably will never change this. It just feels not ideal having this dependency without at least a comment here that this dependency exists.

Reply:
done!